src/main/java/software/amazon/smithy/plugin/language/Smithy.bnf

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+{
+  classHeader = "CopyrightHeader.txt"
+  parserClass="software.amazon.smithy.plugin.parser.SmithyParser"
+
+  extends="com.intellij.extapi.psi.ASTWrapperPsiElement"
+
+  psiClassPrefix="Smithy"
+  psiImplClassSuffix="Impl"
+  psiPackage="software.amazon.smithy.plugin.language.psi"
+  psiImplPackage="software.amazon.smithy.plugin.language.psi.impl"
+
+  elementTypeHolderClass="software.amazon.smithy.plugin.language.psi.SmithyTypes"
+  elementTypeClass="software.amazon.smithy.plugin.language.psi.SmithyElementType"
+  tokenTypeClass="software.amazon.smithy.plugin.language.psi.SmithyTokenType"
+
+  tokens = [
+    DOC_COMMENT = "regexp:///[^\r\n]*"
+    LINE_COMMENT = "regexp://[^\r\n]*"
+    SPACE = "regexp:\s+"
+    NEWLINE = "regexp:\r?\n"
+    VERSION = "$version"
+    APPLY = "apply"
+    METADATA = "metadata"
+    NAMESPACE_KEYWORD = "namespace"
+    USE_KEYWORD = "use"
+    QUOTED_TEXT = "regexp:\"([^\\\"]|\\.)*\""
+    DQUOTE = '"'
+    ESCAPE = '\'
+    COLON = ":"
+    EQ = "="
+    COMMA = ","
+    LBRACE =  "{"
+    RBRACE =  "}"
+    LBRACK =  "["
+    RBRACK =  "]"
+    LPAREN = "("
+    RPAREN = ")"
+    POUND = "#"
+    NODE_CONSTANTS = "regexp:(TRUE|FALSE|NULL)"
+    MINUS = "-"
+    PLUS = "+"
+    TRAIT_ID = "regexp:@(((_|\p{Alpha})(_|\p{Alpha}|\p{Digit})*(.(_|\p{Alpha})(_|\p{Alpha}|\p{Digit})*)*)#)*(_|\p{Alpha})(_|\p{Alpha}|\p{Digit})*"
+    INT = "regexp:0|[1-9][0-9]*"
+    FRAC = "regexp:\.[0-9]*"
+    EXP = "regexp:[eE]([-a+])?[0-9]+"
+    SIMPLE_TYPE_NAME = "regexp:(blob|boolean|document|string|byte|short|integer|long|float|double|bigInteger|bigDecimal|timestamp)"
+    COMPLEX_TYPE_NAME = "regexp:(list|set|map|structure|union)"
+    NODE_OBJECT_TYPE_NAME = "regexp:(service|resource|operation)"
+    SHAPE_ID_MEMBER = "regexp:\$(_|\p{Alpha})(_|\p{Alpha}|\p{Digit})*"
+    IDENTIFIER = "regexp:(_|\p{Alpha})(_|\p{Alpha}|\p{Digit})*"
+    TEXT_BLOCK = "regexp:\"\"\"(\r?\n[ !#-\[\]-}]*)*\"\"\""
+  ]
+}
+
+idl ::= ws version_section ws metadata_section shape_section
+
+version_section ::= VERSION COLON ws QUOTED_TEXT
```

Comment:
If it's not feasible to make this a full `control_section` at the moment, let's leave a note here.